### PR TITLE
llm performance scripts 

### PR DESCRIPTION
**Pull request by malay-nagda**
# What does this PR do ?

Adds scripts for llm pre-training and fine-tuning, optimized for performance

**Collection**: [llm]

# Changelog 

Added arg params for HuggingFace Token, NEMO_HOME env var and fine-tuning scheme

```
-hf/--hf_token # needed for downloading checkpoints and tokenizers from HF
-nh/--nemo_home # needed for accessing locally stored checkpoints and tokenizers
-f/--finetuning # 'lora' or 'sft'. default is 'lora' 
```

# Usage

```python
python3 scripts/llm/performance/pretrain_llama3_8b.py -a <slurm_account> -p <slurm_partition>
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: scripts/llm/performance/utils.py [hunk 1]
```diff
@@ -14,43 +14,49 @@
...
-            "function.",
-        )
-
-    mounts = []
-    if custom_mounts:
-        mounts.extend(custom_mounts)
+    err_msgs = []
+    if log_dir != NEMORUN_HOME:
+        err_msgs.append(f"\nRun `export NEMORUN_HOME={log_dir}` in your shell environment and rerun this script.")
+    if nemo_home != DEFAULT_NEMO_HOME:
```
[the remark this answers is not in the document]
Comment (malay-nagda) [reply scripts/llm/performance/utils.py:55]: This check is redundant, will remove it. 

The default `NEMO_HOME` is `~/.cache/nemo`. If the user wants to set it to a different location, they have to explicitly set it on the compute node or update `SlurmExecutor.env_vars` with this env var, which sets it via the sbatch script.

I think this arg can remove 1 extra step in setting custom `NEMO_HOME` as they can-
1. update `SlurmExecutor.env_vars` in this file or
2. pass it as an arg and avoid the extra step of manually updating this file

**Comment by erhoo82** [comment]
@malay-nagda 
Scripts share many lines of code. Should we create a parent function and have the common part inherited?

**Review by erhoo82 — APPROVED** [review]
